Automattic/wp-calypso
# Plans: Update components/plans module to ECMAScript 6

This pull request updates the `components/plans` section to use as many [ES6](https://github.com/Automattic/calypso-pre-oss/blob/master/docs/coding-guidelines/javascript.md#es6) features as possible. It also fixes and improves a few minor things in the process - such as unnecessary code.
 
![screenshot](https://cloud.githubusercontent.com/assets/594356/14109625/2d87a6e0-f5c3-11e5-8082-76fbb43df23c.png)

#### Additional notes
 
You should probably consider browsing commits [one by one](https://github.com/Automattic/wp-calypso/pull/4394/commits) to review this pull request since it might be easier to understand the changes rather than in the [unified diff view](https://github.com/Automattic/wp-calypso/pull/4394/files).

#### Testing instructions
 
1. Run `git checkout update/components-plans-to-es6` and start your server
2. Check that the [`Plans` page](http://calypso.localhost:3000/plans) behave as before
3. Check that the [`Compare Plans` page](http://calypso.localhost:3000/plans/compare) behave as before
4. Purchase a Premium plan and check the Plans pages again
5. Execute `localStorage.setItem( 'ABTests', '{"freeTrials_20160120":"offered"}' )` in your browser's console
6. Navigate to the `Plans` page again and start a free trial
7. Check that the [`Plan Overview` page](http://calypso.localhost:3000/plans) behave as before

#### Reviews
